Number of compilation errors using `select!` macro

I've tried version "0.3.0-alpha.11" and master at commit 8479bd26d5cafefb91c0d2d0f949cdad38c1dde2.

```rust
// Both these values implement `Future` and `FusedFuture`:
let mut timeout = Timer::timeout(&mut ctx, Duration::from_millis(100)).fuse();
let mut msg_future = ctx.receive().fuse();

select! {
    msg = msg_future => println!("Got a message: {}", msg),
    _ = timeout => {
        println!("Getting impatient!");
        return;
    },
};
```

It causes the following compilation errors.

```log
error[E0433]: failed to resolve: could not find `FusedFuture` in `future`
  --> src/actor/context.rs:87:9
   |
25 | /         select! {
26 | |             msg = msg_future => println!("Got a message: {}", msg),
27 | |             _ = timeout => {
28 | |                 println!("Getting impatient!");
29 | |                 continue;
30 | |             },
31 | |         };
   | |__________^ could not find `FusedFuture` in `future`
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0433]: failed to resolve: could not find `Poll` in `task`
  --> src/actor/context.rs:87:9
   |
25 | /         select! {
26 | |             msg = msg_future => println!("Got a message: {}", msg),
27 | |             _ = timeout => {
28 | |                 println!("Getting impatient!");
29 | |                 continue;
30 | |             },
31 | |         };
   | |__________^ could not find `Poll` in `task`
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0412]: cannot find type `LocalWaker` in module `futures_util::task`
  --> src/actor/context.rs:87:9
   |
25 | /         select! {
26 | |             msg = msg_future => println!("Got a message: {}", msg),
27 | |             _ = timeout => {
28 | |                 println!("Getting impatient!");
29 | |                 continue;
30 | |             },
31 | |         };
   | |__________^ did you mean `LocalWakerRef`?
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
help: possible candidates are found in other modules, you can import them into scope
   |
5  | use core::task::LocalWaker;
   |
5  | use futures_core::task::LocalWaker;
   |
5  | use futures_util::core_reexport::task::LocalWaker;
   |
5  | use std::task::LocalWaker;
   |

error[E0412]: cannot find type `Poll` in module `futures_util::task`
  --> src/actor/context.rs:87:9
   |
25 | /         select! {
26 | |             msg = msg_future => println!("Got a message: {}", msg),
27 | |             _ = timeout => {
28 | |                 println!("Getting impatient!");
29 | |                 continue;
30 | |             },
31 | |         };
   | |__________^ not found in `futures_util::task`
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
help: possible candidates are found in other modules, you can import them into scope
   |
5  | use core::task::Poll;
   |
5  | use futures_core::Poll;
   |
5  | use futures_core::task::Poll;
   |
5  | use futures_util::core_reexport::task::Poll;
   |
and 1 other candidates

error[E0277]: the size for values of type `[_]` cannot be known at compilation time
  --> src/actor/context.rs:87:9
   |
25 | /         select! {
26 | |             msg = msg_future => println!("Got a message: {}", msg),
27 | |             _ = timeout => {
28 | |                 println!("Getting impatient!");
29 | |                 continue;
30 | |             },
31 | |         };
   | |__________^ doesn't have a size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `[_]`
   = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
   = note: all local variables must have a statically known size
   = help: unsized locals are gated as an unstable feature
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

I'm a little lost as to what is wrong with the code. Also the code did work with version "0.3.0-alpha.10", but I rewrote it for alpha 11 and can't figure these errors out.

Any help would be appreciated.

--- a/futures-util/src/future/mod.rs
+++ b/futures-util/src/future/mod.rs
@@ -9,6 +9,10 @@ use futures_core::future::Future;
 use futures_core::stream::Stream;
 use futures_core::task::{LocalWaker, Poll};
 
+// re-export for `select!`
+#[doc(hidden)]
+pub use futures_core::future::FusedFuture;
+
 // Primitive futures
 mod empty;
 pub use self::empty::{empty, Empty};

--- a/futures-util/src/task/mod.rs
+++ b/futures-util/src/task/mod.rs
@@ -16,3 +16,7 @@ pub use self::local_waker_ref::{local_waker_ref, local_waker_ref_from_nonlocal, 
     cfg(all(target_has_atomic = "cas", target_has_atomic = "ptr"))
 )]
 pub use futures_core::task::__internal::AtomicWaker;
+
+// re-export for `select!`
+#[doc(hidden)]
+pub use futures_core::task::{LocalWaker, Poll};

--- a/futures-util/tests/select_next_some.rs
+++ b/futures-util/tests/select_next_some.rs
@@ -55,3 +55,32 @@ fn select() {
         assert_eq!(total, 6);
     });
 }
+
+// Check that `select!` macro does not fail when importing from `futures_util`.
+#[test]
+fn futures_util_select() {
+    use futures_util::select;
+
+    // Checks that even though `async_tasks` will yield a `None` and return
+    // `is_terminated() == true` during the first poll, it manages to toggle
+    // back to having items after a future is pushed into it during the second
+    // poll (after pending_once completes).
+    futures::executor::block_on(async {
+        let mut fut = future::ready(1).pending_once();
+        let mut async_tasks = FuturesUnordered::new();
+        let mut total = 0;
+        loop {
+            select! {
+                num = fut => {
+                    total += num;
+                    async_tasks.push(async { 5 });
+                },
+                num = async_tasks.select_next_some() => {
+                    total += num;
+                }
+                complete => break,
+            }
+        }
+        assert_eq!(total, 6);
+    });
+}

